core/tests/behavior/capability.rs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
We don't need to add this behavior test since they already been covered by other tests.

Reply:
I found many tests don’t cover the capability checks. For example, presign tests will check the capability in their preludes, and just skip the test body if it’s not supported. To ensure the check is working for users, we need to simulate the use cases where users blindly perform some unsupported operations without checking the capability first.

Reply:
> I found many tests don’t cover the capability checks. For example, presign tests will check the capability in their preludes, and just skip the test body if it’s not supported.

Yes, there are part of our coming behavior tests refactor.

Reply:
> To ensure the check is working for users, we need to simulate the use cases where users blindly perform some unsupported operations without checking the capability first.

Those logic are so simple and direct that I don't think we should add behavior test for them. We can add some unit tests in `CompleteLayer` instead.

Reply:
That sounds like a good idea.